include/GIL/Instructions/PtrOffsetInst.hpp

```diff
@@ -0,0 +1,59 @@
+#ifndef GLU_GIL_INSTRUCTIONS_PTR_OFFSET_INST_HPP
+#define GLU_GIL_INSTRUCTIONS_PTR_OFFSET_INST_HPP
+
+#include "AggregateInst.hpp"
+
+namespace glu::gil {
+
+/// @class PtrOffsetInst
+/// @brief Represents a pointer offset instruction in GIL.
+///
+/// This instruction calculates a new pointer by applying an integer offset to a
+/// base pointer. The result is a new pointer of the same type.
+class PtrOffsetInst : public AggregateInst {
+    Value basePtr;
+    Value offset;
+
+public:
+    /// @brief Constructs a PtrOffsetInst object.
+    ///
+    /// @param basePtr The base pointer to offset from.
+    /// @param offset The integer offset to apply.
+    PtrOffsetInst(Value basePtr, Value offset)
+        : AggregateInst(InstKind::PtrOffsetInstKind)
+        , basePtr(basePtr)
+        , offset(offset)
+    {
+        // TODO: assert(llvm::isa<glu::types::PointerTy>(*value.getType()));
```

Comment:
BasePtr is PointerTy and offset is IntTy. I don’t think you need it commented anymore

Reply:
Assert that offset is an IntTy